python/pyarrow/array.pyx

```diff
@@ -45,16 +46,40 @@ cdef _pandas():
     return pd
 
 
+cdef maybe_coerce_datetime64(values, dtype, DataType type,
```

Comment:
Since there's `__builtin__.type` I haven't worried too much about it, but I'm OK with another naming convention, like `type_`, what do you prefer?